source/main/main.cpp

```diff
@@ -367,6 +367,7 @@ int main(int argc, char *argv[])
 
                 case MSG_NET_CONNECT_PROGRESS:
                     App::GetGuiManager()->GetLoadingWindow()->SetProgressNetConnect(m.description);
+                    App::GetGuiManager()->SetVisible_GameMainMenu(false);

```

Comment:
This is a hack, there's multiple _PROGRESS messages during connecting and no reason for main menu to appear. I'll investigate it why it appears at all.

Update: right, here's the actual problem: https://github.com/RigsOfRods/rigs-of-rods/blob/d129ba5a7df15f2d97fcc1a497ec36c41ce34771/source/main/gui/panels/GUI_MultiplayerSelector.cpp#L362-L365